vscode/src/projectSystem.ts

```diff
@@ -137,6 +138,22 @@ async function readFileUri(
   const uri: vscode.Uri = (maybeUri as any).path
     ? (maybeUri as vscode.Uri)
     : vscode.Uri.parse(maybeUri as string);
+
+  // If any open documents match this uri, return their contents instead of from disk
+  const opendoc = vscode.workspace.textDocuments.find(
+    (opendoc) =>
+      opendoc.uri.scheme === uri.scheme &&
+      opendoc.uri.authority === uri.authority &&
+      opendoc.uri.path == uri.path,
```

Comment:
Is there a subtle reason this is a double equals while the other ones are triple?

Reply:
Yes, because I'm an idiot. (I guess that's not that subtle). Thanks for the catch! (Surprised ESLint didn't flag it).

In reality, I could probably just do `opendoc.toString() === uri.toString()`, but I'm not sure if query and fragment might have values at some point that should be ignored, so these seems like the right fields of a URI to worry about for comparison.

Reply:
I've been doing the `uri1.toString() === uri2.toString()` comparison elsewhere in the code, and it hasn't caused me trouble so far. But I think this is good too. Surprised they didn't put in an `equals` method.

Yeah, ESLint has a rule for triple equals (`eqeqeq`) but it's not included in the recommended set.

I personally have conditioned myself to *always* use `===` since I don't trust myself to remember when it's okay to use `==`, it always surprises me. I wouldn't mind enabling the lint rule to enforce this, but that discussion is for another time...

Reply:
Switched to comparing the toString() results.